LoRaEngine/modules/LoRaWanNetworkSrvModule/LoraTools/LoRaMessage/LoRaPayload.cs

```diff
@@ -124,7 +125,10 @@ public static byte[] CalculateKey(LoRaPayloadKeyType keyType, byte[] appnonce, b
             aes.Padding = PaddingMode.None;
             aes.IV = new byte[16];
 
-            var pt = type.Concat(appnonce).Concat(netid).Concat(devnonce).Concat(new byte[7]).ToArray();
+            var devNonceBytes = new byte[DevNonce.Size];
+            _ = devnonce.Write(devNonceBytes);
+
+            var pt = type.Concat(appnonce).Concat(netid).Concat(devNonceBytes).Concat(new byte[7]).ToArray();
```

Comment:
What about simplifying this to the following that doesn't require as many allocations and conversion and would allow direct use of `DevNonce.Write` without going through a two-byte array?

```c#
var pt = new byte[appnonce.Length + netid.Length + DevNonce.Size + 7];
Array.Copy(appnonce, pt, appnonce.Length);
Array.Copy(netid, 0, pt, appnonce.Length, netid.Length);
_ = devnonce.Write(pt.AsSpan(appnonce.Length + netid.Length));
```

Reply:
Done in c6e00385d1a8b7384e1e813a8c8d64b343160f05